src/lib/controllers/config-manifest/config-manifest.ts

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright 2019 Balena Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as Ajv from 'ajv';
+import * as _ from 'lodash';
+
+import { loadFromUri } from '../../tools';
+import {
+	ConfigManifestSchema,
+	ConfigManifestKeys,
+} from './config-manifest-schema';
+import { path } from 'temp';
+import { fs } from 'mz';
+import { type } from 'os';
+import { Dictionary } from 'lodash';
+// tslint:disable-next-line:no-var-requires
+const configManifestSchemaJson = require('../../../../schemas/config-manifest-schema.json');
+
+const ajv = new Ajv();
+const configManifestSchema = ajv.compile(configManifestSchemaJson);
+
+/**
+ * ConfigManifest class
+ * Used for transforming/exposing a config-manifest
+ */
+// WIP notes: Kept this only because it provides the ability to return either
+// 	a JSON schema or valid ConfigManifest at this point
+export type NameValueCollection = { [propName: string]: any };
```

Comment:
```suggestion
export type NameValueCollection = Dictionary<any>;
```

Reply:
This wasn't being used in the end, so I dropped it out 👍